docs/developer-guide.md

```diff
@@ -0,0 +1,280 @@
+repo. The pipeline GitHub bot account has permissions to manage it.
+
+> **Note:** This repository is only configured for testing certified operators,
+> NOT Red Hat Marketplace operators (see
+> [`config.yaml`](https://github.com/redhat-openshift-ecosystem/operator-pipelines-test/blob/main/config.yaml)).
+
+```bash
+# Checkout the pipelines test repo
+git clone https://github.com/redhat-openshift-ecosystem/operator-pipelines-test
+cd operator-pipelines-test
+
+# Create a new branch
+git checkout -b <insert-branch-name>
+
+# Copy your package directory
+cp -R <package-dir>/ operators/
+
+# Commit changes
+git add -A
+
+# Use this commit pattern so it defaults to the pull request title.
+# This is critical to the success of the pipelines.
+git commit -m "operator <operator-package-name> (<bundle-version>)"
+
+# Push your branch. Open the pull request using the output.
+git push origin <insert-branch-name>
+```
+
+> **Note:** You may need to merge the pull request to use it for testing the release pipeline.
+
+#### Create an API Key
+
+Contact a Pyxis admin to assist with this request. It must correspond to the
+`org_id` for the certification project under test.
+
+## Making Changes to the Pipelines
+
+### Guiding Principles
+
+- Avoid including credentials within Task scripts.
+  - Avoid the use of `set -x` in shell scripts which _could_ expose credentials to the console.
+- Don't use workspaces for passing secrets. Use `secretKeyRef` and `volumeMount`
+  with secret and key names instead.
+  - Reason: It adds unnecessary complexity to `tkn` commands.
+- Use images from trusted registries/namespaces.
+  - registry.redhat.io
+  - registry.access.redhat.com
+  - quay.io/redhat-isv
+  - quay.io/opdev
+- Use image pull specs with digests instead of tags wherever possible.
+- Tasks must implement their own skipping behavior, if needed.
+  - Reason: If a task is not executed, any dependent tasks will not be executed either.
+- Don't use ClusterTasks or upstream tasks. All tasks are defined in this repo.
+- Document all params, especially pipeline params.
+- Output human readable logs.
+- Use reasonable defaults for params wherever possible.
+
+### Applying Pipeline Changes
+
+You can use the following command to apply all local changes to your OCP project.
+It will add all the Tekton resources used across all the pipelines.
+
+```bash
+oc apply -R -f ansible/roles/operator-pipeline/templates/openshift
+```
+
+## Making Changes to the Pipeline Image
+
+### Setup
+To install the python package in a development environment, run:
+
+```bash
+pip install ".[dev]"
+```
+
+### Tips
+
+- If adding a new script - don't forget to add the entrypoint to setup.py
+
+### Run Unit Tests, Code Style Checkers, etc.
+
+To run unit tests and code style checkers:
+
+```bash
+tox
+```
+
+### Build & Push
+
+1. Ensure you have [buildah](https://github.com/containers/buildah/blob/main/install.md) installed
+
+1. Build the image
```

Comment:
```suggestion
2. Build the image
```

Reply:
This is intentional. It's a [markdown feature](https://www.markdownguide.org/basic-syntax/#ordered-lists) so that you don't have to maintain ordered list numbers.

Reply:
Ooh TIL